src/components/signup/SignUp.js

```diff
@@ -2,29 +2,56 @@ import "./SignUp.css";
 import { useState, useEffect } from "react";
 import axios from "axios";
 import { useAuth } from "../../contexts/auth-context";
+import { validSignUp } from "../../utilities/auth";
 
 const SignUp = () => {
   const { authDispatch } = useAuth();
 
-  const [userInfo, setUserInfo] = useState({
-    firstname: "",
-    secondname: "",
+  const [signUpData, setSignUpData] = useState({
+    firstName: "",
+    lastName: "",
     mobile: "",
     email: "",
     password: "",
+    confirmPassword: "",
+    terms: false,
   });
 
-  const inputHandler = (e) => {
-    e.preventDefault();
+  const [signUpErrors, setSignUpErrors] = useState({
+    firstName: "",
+    lastName: "",
+    mobile: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+    terms: "",
+  });
 
-    setUserInfo((data) => ({ ...data, [e.target.name]: e.target.value }));
+  const inputHandler = (e) => {
+    if (e.target.value === "checkbox") {
+      setSignUpData((signUpData) => ({
+        ...signUpData,
+        [e.target.id]: e.target.checked,
+      }));
+    }
+    setSignUpData((data) => ({ ...data, [e.target.name]: e.target.value }));
+    setSignUpErrors((errData) => ({
+      ...errData,
+      [e.target.name]: "",
+    }));
   };
-  const postSignUpDetails = async (e) => {
-    e.preventDefault();
-    authDispatch({ type: "USER_LOAD" });
 
+  console.log(signUpData);
+  const postSignUpDetails = async (e) => {
     try {
-      const { data } = await axios.post("/api/auth/signup", userInfo);
+      e.preventDefault();
+      const { isValid, errors } = validSignUp(signUpData, signUpErrors);
+      if (!isValid) {
+        setSignUpErrors(errors);
+        return;
```

Comment:
What's the use of return here?

Reply:
if This conditions is true I will just return then and there , code will not run further .